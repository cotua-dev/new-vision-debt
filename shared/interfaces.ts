export interface PageProps {
    props: {};
    revalidate: number;
};
