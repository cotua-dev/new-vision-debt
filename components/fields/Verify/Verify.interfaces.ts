import { Dispatch, SetStateAction } from 'react';
import { Step } from '../../Stepper/Stepper.interfaces';

export interface VerifyProps {
    steps: Step[];
    currentStep: number;
    setDisableNextButton: Dispatch<SetStateAction<boolean>>;
    code: string;
    setCode: Dispatch<SetStateAction<string>>;
    disableVerifyField: boolean;
};
