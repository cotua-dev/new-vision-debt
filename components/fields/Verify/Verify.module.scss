@use 'styles/mixins';
@use 'styles/variables';

div.code-field-wrapper {
    @include mixins.flexify(center, center, column);

    input.code-field {
        @include mixins.base-field;
    }

    small.error {
        @include mixins.error;
    }

    small.note {
        @include mixins.note;

        a {
            color: variables.$secondary-color;
        }
    }
}
