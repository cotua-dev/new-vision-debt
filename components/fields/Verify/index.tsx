import Link from 'next/link';
import { ChangeEvent, useState } from 'react';
import NumberFormat from 'react-number-format';
import { VerifyProps } from './Verify.interfaces';
import styles from './Verify.module.scss';

export function Verify(props: VerifyProps): JSX.Element {
    const { steps, currentStep, setDisableNextButton, code, setCode, disableVerifyField } = props;

    // Should match a code of exactly 6 numbers between 0 and 9
    const codeRegex: RegExp = new RegExp(/^[0-9]{6}$/);

    const [error, setError] = useState('');
    const [showError, setShowError] = useState(false);

    /**
     * Set the value of code for the step and `code`
     * @param value String of the code
     */
    function handleCodeSetting(value: string): void {
        // Set `code`
        setCode(value);

        // Check to see if we pass the regular expression test
        if (codeRegex.test(value)) {
            // Set and remove error
            setError('');
            setShowError(false);

            // Set the current step's `phone` value
            steps[currentStep].value.phone = value;

            // Make the current step valid
            steps[currentStep].validity = true;

            // Enable the next button
            setDisableNextButton(false);
        } else {
            // Set and show error
            setError('Please provide a valid code');
            setShowError(true);

            // Disable the next button
            setDisableNextButton(true);
        }
    }

    return (
        <div className={styles['code-field-wrapper']}>
            <NumberFormat
                className={styles['code-field']}
                format="######"
                type="text"
                id="code"
                name="code"
                value={code}
                onChange={(event: ChangeEvent<HTMLInputElement>) => handleCodeSetting(event.target.value)}
                disabled={disableVerifyField}
            />
            <small className={styles['note']}>
                {`By clicking the "Call or Text Me Now‚Äù button below, you agree to receive promotional messages sent via an autodialer and other automated systems. This agreement isn't a condition of any purchase. Terms and Privacy Policy can be found at `}
                <Link href="/terms-of-use"><a aria-label="Terms of Use Page Link">{`Terms`}</a></Link>
                {` and `}
                <Link href="/privacy-policy"><a aria-label="Privacy Policy Page Link">{`Privacy Policy.`}</a></Link>
                {` You may receive every week until you tell us to stop trying to contact you. To stop receiving phone calls, call `}
                <a href="tel:+18886838681">{`888-683-8681.`}</a>
                {` To stop receiving text messages, reply STOP. To stop receiving emails, click the UNSUBSCRIBE button in the message.`}
            </small>
            {
                showError &&
                <small className={styles['error']}>{error}</small>
            }
        </div>
    );
}
