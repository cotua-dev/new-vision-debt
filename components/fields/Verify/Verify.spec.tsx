import { render, RenderResult } from '@testing-library/react';
import { Verify } from './index';
import { VerifyProps } from './Verify.interfaces';

describe('Verify Component', (): void => {
    let renderResult: RenderResult;
    const props: VerifyProps = {
        steps: [],
        currentStep: 0,
        setDisableNextButton: () => {},
        code: '',
        setCode: () => {},
        disableVerifyField: false,
    };

    beforeAll((): RenderResult => renderResult = render(<Verify {...props}/>));

    test('renders', (): void => {
        const { container } = renderResult;
        expect(container.firstChild).toMatchSnapshot();
    });

    afterAll((): void => renderResult.unmount());
});
