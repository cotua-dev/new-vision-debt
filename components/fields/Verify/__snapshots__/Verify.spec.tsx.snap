// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify Component renders 1`] = `
<div
  class="code-field-wrapper"
>
  <input
    class="code-field"
    id="code"
    inputmode="numeric"
    name="code"
    type="text"
    value=""
  />
  <small
    class="note"
  >
    By clicking the "Call or Text Me Now‚Äù button below, you agree to receive promotional messages sent via an autodialer and other automated systems. This agreement isn't a condition of any purchase. Terms and Privacy Policy can be found at 
    <a
      aria-label="Terms of Use Page Link"
      href="/terms-of-use"
    >
      Terms
    </a>
     and 
    <a
      aria-label="Privacy Policy Page Link"
      href="/privacy-policy"
    >
      Privacy Policy.
    </a>
     You may receive every week until you tell us to stop trying to contact you. To stop receiving phone calls, call 
    <a
      href="tel:+18009211420"
    >
      800-921-1420.
    </a>
     To stop receiving text messages, reply STOP. To stop receiving emails, click the UNSUBSCRIBE button in the message.
  </small>
</div>
`;
